### DISCUSSION TITLE
Use spqr-specific (patched) PostgreSQL version

### DISCUSSION THREAD

**Discussion by reshke** [pg-sharding/spqr#373] [Ideas]
There are various reasons to use __better__ version of PostgreSQL to run in cloud environments. 

For spqr, number of improvements needed to provide some features of unsharded base-metal PostgreSQL instance 

1.) Cloud service auth feature to allow SCRAM auth in proxy-mode (to perform SCRAM backend auth under specified role without its plain text password), see [commit](https://github.com/pg-sharding/cpg/commit/3d61fbbaebe8a801a64b6eb79da5f3ce07683730)

2.) Cluster-wide snapshots and intelligent cross-shard queries support using Commit Sequence Number patch (rebased last version from pgsql-hackers on PG_16), refer to [pr](https://github.com/pg-sharding/cpg/pull/1)

https://github.com/pg-sharding/cpg

**Comment by Denchick** [comment]
Hi! Nice idea but we have to provide backward compatibility with upstream postgres

**Reply by reshke** [reply]
no we dont
we can use feature flag to resolve if we use patched/unpatched version
